src/preferenceCenter/preferenceCenter.commands.js

```diff
@@ -0,0 +1,183 @@
+
+export default class PreferenceCenterCommands {
+  editor;
+
+  dcPopup;
+
+  pcService;
+
+  logger;
+
+  constructor(editor) {
+    this.pcService = new PreferenceCenterService(editor);
+    this.logger = new Logger(editor);
+  }
+
+  launchPreferenceCenterPopup(editor, sender, options) {
+    // Transform DC slots to token
+    // editor.runCommand('preset-mautic:dynamic-content-slots-to-tokens');
+
+    this.showPreferenceCenterEditor(editor, options);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  stopPreferenceCenterPopup(editor) {
+    editor.runCommand('preset-mautic:dynamic-content-tokens-to-slots');
+  }
+
+  /**
+   * Convert all dynamic content tokens on the canvas to
+   * human readable texts/slots/component.
+   * {dynamiccontent} => Dynamic Content
+   */
+  convertPreferenceCenterTokensToSlots() {
+    const components = this.pcService.getDcComponents();
+
+    components.forEach((comp) => {
+      this.pcService.transformDcTokenToSlot(comp);
+    });
+  }
+
+  /**
+   * Convert dynamic content slots to tokens
+   * Dynamic Content => {preferenceCenter}
+   *
+   * @param editor
+   * @returns integer nr of dynamic content slots found
+   */
+  // eslint-disable-next-line class-methods-use-this
+  convertPreferenceCenterSlotsToTokens(editor) {
+    // get all dynamic content elements loaded in the editor
+    const preferenceCenter = editor.DomComponents.getWrapper().find('[data-slot="preferenceCenter"]');
+    preferenceCenter.forEach((preferenceCenter) => {
+      const attributes = preferenceCenter.getAttributes();
+      const decId = attributes['data-param-dec-id'];
+      // If it's not a token -> convert to token
+      if (decId >= 0) {
+        const dynConId = PreferenceCenterCommands.getPcStoreId(attributes['data-param-dec-id']);
+
+        const dynConTarget = mQuery(dynConId);
+        const dynConName = dynConTarget.find(`${dynConId}_tokenName`).val();
+        const dynConToken = `{dynamiccontent="${dynConName}"}`;
+
+        // Clear id because it's reloaded by Mautic and this prevent slot to be destroyed by GrapesJs destroy event on close.
+        // preferenceCenter.addAttributes({ 'data-param-dec-id': '' });
+        preferenceCenter.set('content', dynConToken);
+      }
+    });
+
+    return preferenceCenter.length;
+  }
+
+  /**
+   * Build and display the Dynamic Content editor popup/modal window.
+   * @param {Model} component The current grapesjs component
+   */
+  // eslint-disable-next-line class-methods-use-this
+  showPreferenceCenterEditor(editor, options) {
+    this.dcPopup = PreferenceCenterCommands.buildPreferenceCenterPopup();
+
+    this.addPreferenceCenterEditor(editor, options);
+
+    const title = Mautic.translate('grapesjsbuilder.preferenceCenterBlockLabel');
+    const modal = editor.Modal;
+    modal.setTitle(title);
+
+    editor.Modal.setContent(this.dcPopup);
+    modal.open();
+    modal.onceClose(() => editor.stopCommand('preset-mautic:dynamic-content-open'));
```

Comment:
I guess this should stop the preference center modal, not the dynamic content modal. This could be the fix for the problem that you can not open the modal multiple times.